From bd4bbeee432773ce94854c5a0f3ff99307a205ad Mon Sep 17 00:00:00 2001
From: Marcelo Lira <marcelo.lira@openbossa.org>
Date: Fri, 6 Aug 2010 16:50:23 -0300
Subject: [PATCH] Modifies cmake file to workaround a problem in QtMobility package.

QtMobility installation is missing the file
/usr/share/qt4/mkspecs/features/mobilityconfig.prf on Fremantle.
Modifications were made to the cmake build files to fix the problem.
---
 cmake/Modules/FindQtMobility.cmake |   15 +++++++++++----
 cmake/Modules/mobilityconfig.prf   |    6 ++++++
 2 files changed, 17 insertions(+), 4 deletions(-)
 create mode 100644 cmake/Modules/mobilityconfig.prf

diff --git a/cmake/Modules/FindQtMobility.cmake b/cmake/Modules/FindQtMobility.cmake
index ad153dc..444caa0 100644
--- a/cmake/Modules/FindQtMobility.cmake
+++ b/cmake/Modules/FindQtMobility.cmake
@@ -1,8 +1,15 @@
 INCLUDE(FindQt4)
 
+set(MOBILITY_CONFIG_MKSPECS_FILE "")
+IF(EXISTS "${QT_MKSPECS_DIR}/features/mobilityconfig.prf")
+    set(MOBILITY_CONFIG_MKSPECS_FILE "${QT_MKSPECS_DIR}/features/mobilityconfig.prf")
+ELSEIF(EXISTS "${QT_MKSPECS_DIR}/features/mobility.prf")
+    set(MOBILITY_CONFIG_MKSPECS_FILE "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/mobilityconfig.prf")
+ENDIF()
+
 macro(export_component component)
-    IF(EXISTS "${QT_MKSPECS_DIR}/features/mobilityconfig.prf")
-        FILE(READ ${QT_MKSPECS_DIR}/features/mobilityconfig.prf MOBILITY_FILE_CONTENTS)
+    IF(NOT ${MOBILITY_CONFIG_MKSPECS_FILE} STREQUAL "")
+        FILE(READ ${MOBILITY_CONFIG_MKSPECS_FILE} MOBILITY_FILE_CONTENTS)
         STRING(TOLOWER ${component} _COMPONENT)
         IF(${MOBILITY_FILE_CONTENTS} MATCHES "MOBILITY_CONFIG=.*${_COMPONENT}.*")
             STRING(TOUPPER ${component} _COMPONENT)
@@ -30,8 +37,8 @@ IF(EXISTS "${QT_MKSPECS_DIR}/features/mobility.prf")
     SET(QT_MOBILITY_LIBRARY_DIR ${CMAKE_MATCH_1})
 
     #VERSION
-    iF(EXISTS "${QT_MKSPECS_DIR}/features/mobilityconfig.prf")
-        FILE(READ ${QT_MKSPECS_DIR}/features/mobilityconfig.prf MOBILITY_CONFIG_FILE_CONTENTS)
+    IF(NOT ${MOBILITY_CONFIG_MKSPECS_FILE} STREQUAL "")
+        FILE(READ ${MOBILITY_CONFIG_MKSPECS_FILE} MOBILITY_CONFIG_FILE_CONTENTS)
         STRING(REGEX MATCH "MOBILITY_VERSION = ([^\n]+)" QT_MOBILITY_VERSION "${MOBILITY_CONFIG_FILE_CONTENTS}")
         SET(QT_MOBILITY_VERSION ${CMAKE_MATCH_1})
 
diff --git a/cmake/Modules/mobilityconfig.prf b/cmake/Modules/mobilityconfig.prf
new file mode 100644
index 0000000..794c95b
--- /dev/null
+++ b/cmake/Modules/mobilityconfig.prf
@@ -0,0 +1,6 @@
+MOBILITY_CONFIG=bearer location contacts multimedia publishsubscribe versit messaging systeminfo serviceframework sensors
+MOBILITY_VERSION = 1.0.2
+MOBILITY_MAJOR_VERSION = 1
+MOBILITY_MINOR_VERSION = 0
+MOBILITY_PATCH_VERSION = 2
+
-- 
1.7.0.4

